// Copyright 2025 Dennis Hezel
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package test.v1;

import "test/msg/message.proto";

service Test {
  rpc ServerStreaming(msg.Request) returns (stream msg.Response) {}

  rpc ClientStreaming(stream msg.Request) returns (msg.Response) {}

  rpc BidirectionalStreaming(stream msg.Request) returns (stream msg.Response) {
  }

  rpc Unary(msg.Request) returns (msg.Response) {}

  rpc Subscribe(stream Topic) returns (stream Feed) {}
}

message Topic {
  int32 id = 1;
}

message Feed {
  string content = 1;
}